src/main/groovy/de/felixschulze/gradle/HockeyAppUploadTask.groovy

```diff
@@ -102,23 +105,26 @@ class HockeyAppUploadTask extends DefaultTask {
             throw new IllegalArgumentException("Cannot upload to HockeyApp because API Token is missing")
         }
 
-        if (!applicationFile?.exists()) {
-            if (applicationFile) {
-                logger.debug("App file doesn't exist: "+applicationFile?.absolutePath)
-            }
+        if (!applicationFiles) {
             if (!applicationVariant && !hockeyApp.appFileNameRegex) {
                 throw new IllegalArgumentException("No appFileNameRegex provided.")
             }
             if (!hockeyApp.outputDirectory || !hockeyApp.outputDirectory.exists()) {
                 throw new IllegalArgumentException("The outputDirectory (" + hockeyApp.outputDirectory ? hockeyApp.outputDirectory.absolutePath : " not defined " + ") doesn't exists")
             }
-            applicationFile = FileHelper.getFile(hockeyApp.appFileNameRegex, hockeyApp.outputDirectory);
-            if (!applicationFile) {
+            applicationFiles = FileHelper.getFiles(hockeyApp.appFileNameRegex, hockeyApp.outputDirectory);
+            if (!applicationFiles) {
                 throw new IllegalStateException("No app file found in directory " + hockeyApp.outputDirectory.absolutePath)
             }
         }
 
-        logger.lifecycle("App file: " + applicationFile.absolutePath)
+        def appFilePaths = applicationFiles.collect { it.name }
+
+        if (!hockeyApp.allowMultipleAppFiles && applicationFiles.size() > 1) {
```

Comment:
Ok. Thank you.

Btw: yesterday I've got some very strange issue. After integration of the my version of the plugin the instrumentation tests in our project start failing. I don't see any connection between hockeyapp plugin and instrumentation tests, but that's it - if I use plugin from maven central, everything working ok. If I use my version, instrumentation tests start failing (some problems with running of the test tasks).

I decided that this is a problem with new Android gradle plugin (1.5.0), that used in your master branch. So I started a new branch from your 3.3.1 tag, cherry-picked my commits to it and build new plugin. But instrumentation tests still failing. I'm really confused by this. Trying to investigate the issue.